test/requests/resolve_test_commands_test.rb

```diff
@@ -1132,5 +1132,64 @@ def test_ignores_items_tagged_by_other_frameworks
         assert_empty(result[:commands])
       end
     end
+
+    def test_addons_can_support_other_frameworks
+      Class.new(RubyLsp::Addon) do
+        def activate(global_state, outgoing_queue); end
+
+        def name
+          "Rails add-on"
+        end
+
+        def deactivate; end
+
+        def version
+          "0.1.0"
+        end
+
+        def resolve_test_commands(items)
+          example = items.dig(0, :children, 0)
```

Comment:
Are addons expected to go through all items and all their children (and their children) to filter out unrelated items, and then generate commands from the remaining ones?

Reply:
Yes. However, after #3349, the top level items will already be tagged with the framework they are related to, so add-ons can exclude them before going through the children.